Mccalluc/user file browser real data

mccalluc opened the pull request:
I'm trying to take the existing code for the file browser 2, identify just the methods I need, and test them. Tests fail right now: The test does not see the factory it should be testing. I also need to figure out how to run these tests locally.

jkmarx commented:
https://github.com/refinery-platform/refinery-platform/wiki/Client-&-UI-development#grunt-tasks
@mccalluc Running local ui tests

mccalluc commented:
Not sure what this guideline implies for me:
> For consistency, use factories instead of services, since they are so similar.

I have a `services` directory, which contains `factory.js` and `user-file.js`, the latter of which is actually a factory:
```javascript
.factory('userFileService', [...
```
- Should the directory be named something different?
- Should the `factory.js` have a more descriptive name, if everything is a factory?

jkmarx commented on refinery/ui/source/js/user-file-browser/services/factory.js:
```diff
@@ -0,0 +1,79 @@
+      return userFile.$promise;
+    }
+
+    // TODO: I think I'll need this soon-ish.
```
Avoid adding commented code.

jkmarx commented on refinery/ui/source/js/user-file-browser/services/factory.spec.js:
```diff
@@ -0,0 +1,119 @@
+        });
+
+        spyOn(nodeService, 'query').and.callFake(function () {
+          var deferred = $q.defer();
```
why mock nodeService, if you're not using it

jkmarx commented on refinery/ui/source/js/user-file-browser/services/user-file.js:
```diff
@@ -0,0 +1,17 @@
+'use strict';
```
wrap in iffy, according to style guide

jkmarx replied:
other styling issues....see common service tool_definitions for reference

jkmarx replied:
or the tools service

mccalluc commented:
@jkmarx : I think I've addressed all your comments. I've also gotten rid of the mock data in the grid, and am filling in some of the columns with data from solr. TODOs:
- Missing columns
- Column sorting
- CSV
- Functional facets
Not planning to do infinite scrolling, since it sounds like it was hard, and it'd not necessary for this story: They want to be able to narrow it down to just the ones of interest.

jkmarx commented on refinery/ui/source/js/user-file-browser/services/user-file.spec.js:
```diff
@@ -0,0 +1,197 @@
+'use strict';
```
iffy

mccalluc commented:
(Talked with Jennifer and with the iife, the last block is resolved.)
